Sources/Build/Command.compile(ClangModule).swift

```diff
@@ -45,32 +39,91 @@ extension Command {
                 includeFlag = "-iquote"
             }
             args += [includeFlag, dep.path]
-            args += ["-l\(dep.c99name)"] //FIXME: giving path to other module's -fmodule-map-file is not linking that module
         }
+        return args
+    }
+
+    var linkFlags: [String] {
+        var args: [String] = []
+        for case let dep as ClangModule in dependencies {
+            args += ["-l\(dep.c99name)"]
+        }
+        return args
+    }
 
+    func optimizationFlags(conf: Configuration) -> [String] {
         switch conf {
         case .Debug:
-            args += ["-g", "-O0"]
+            return ["-g", "-O0"]
         case .Release:
-            args += ["-O2"]
+            return ["-O2"]
+        }
+    }
+}
+
+private extension Sources {
+    func compilePathsForBuildDir(wd: String) -> [(filename: String, source: String, object: String, deps: String)] {
+        return relativePaths.map { source in
+            let path = Path.join(root, source)
+            let object = Path.join(wd, "\(source).o")
+            let deps = Path.join(wd, "\(source).d")
+            return (source, path, object, deps)
         }
+    }
+}
+
+extension Command {
+    static func compile(clangModule module: ClangModule, configuration conf: Configuration, prefix: String) -> ([Command], Command) {
+
+        let wd = Path.join(prefix, "\(module.c99name).build")
+        let mkdir = Command.createDirectory(wd)
+
+        ///------------------------------ Compile -----------------------------------------
+        var compileCommands = [Command]()
+        let dependencies = module.dependencies.map{ $0.targetName }
+        let basicArgs = module.basicArgs + module.includeFlags + module.optimizationFlags(conf)
+        for path in module.sources.compilePathsForBuildDir(wd) {
+            var args = basicArgs
+            args += ["-MMD", "-MT", "dependencies", "-MF", path.deps]
+            args += ["-c", path.source, "-o", path.object]
+
+            let node = "<\(module.name).\(path.filename)>"
+
+            let clang = ClangTool(desc: "Compiling \(module.name) \(path.filename)",
+                                  inputs: dependencies + [path.source, mkdir.node],
+                                  outputs: [path.object, node],
+                                  args: args,
+                                  deps: path.deps)
+
+            let command = Command(node: node, tool: clang)
+
+            compileCommands.append(command)
+        }
+
+
+        ///FIXME: This probably doesn't belong here
```

Comment:
C Targets require a dylib to be used as modules, so indeed, in this case a product is implicit.

It may be better to still only generate `.o`s here and be explicit with making the product in the products loop in `describe()`.

I haven't strong feelings right now.